### ISSUE TITLE
BUG: `TypeError` when importing `interpolate`

### ISSUE THREAD

**Issue by rees-c** [scipy/scipy#21014]
### Describe your issue.

Trying to import `scipy.stats` raises a `TypeError`.

### Reproducing Code Example

```python
import scipy.stats
```

### Error message

```shell
Traceback (most recent call last):
  File "/Users/reeschang/PycharmProjects/InverseXRD/raw_xrd_simulation.py", line 244, in <module>
    import scipy.stats
  File "/Users/reeschang/miniconda3/envs/ml4pxrds/lib/python3.9/site-packages/scipy/stats/__init__.py", line 606, in <module>
    from ._stats_py import *
  File "/Users/reeschang/miniconda3/envs/ml4pxrds/lib/python3.9/site-packages/scipy/stats/_stats_py.py", line 49, in <module>
    from . import distributions
  File "/Users/reeschang/miniconda3/envs/ml4pxrds/lib/python3.9/site-packages/scipy/stats/distributions.py", line 11, in <module>
    from . import _discrete_distns
  File "/Users/reeschang/miniconda3/envs/ml4pxrds/lib/python3.9/site-packages/scipy/stats/_discrete_distns.py", line 10, in <module>
    from scipy.interpolate import interp1d
  File "/Users/reeschang/miniconda3/envs/ml4pxrds/lib/python3.9/site-packages/scipy/interpolate/__init__.py", line 167, in <module>
    from ._interpolate import *
  File "/Users/reeschang/miniconda3/envs/ml4pxrds/lib/python3.9/site-packages/scipy/interpolate/_interpolate.py", line 14, in <module>
    from . import _fitpack_py
  File "/Users/reeschang/miniconda3/envs/ml4pxrds/lib/python3.9/site-packages/scipy/interpolate/_fitpack_py.py", line 8, in <module>
    from ._fitpack_impl import bisplrep, bisplev, dblint  # noqa: F401
  File "/Users/reeschang/miniconda3/envs/ml4pxrds/lib/python3.9/site-packages/scipy/interpolate/_fitpack_impl.py", line 103, in <module>
    'iwrk': array([], dfitpack_int), 'u': array([], float),
TypeError
```

### SciPy/NumPy/Python version and system information

```shell
1.13.1 1.26.4 sys.version_info(major=3, minor=9, micro=19, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: pkgconfig
    found: true
    include directory: /Users/reeschang/miniconda3/envs/ml4pxrds/include
    lib directory: /Users/reeschang/miniconda3/envs/ml4pxrds/lib
    name: blas
    openblas configuration: unknown
    pc file directory: /Users/reeschang/miniconda3/envs/ml4pxrds/lib/pkgconfig
    version: 3.9.0
  lapack:
    detection method: pkgconfig
    found: true
    include directory: /Users/reeschang/miniconda3/envs/ml4pxrds/include
    lib directory: /Users/reeschang/miniconda3/envs/ml4pxrds/lib
    name: lapack
    openblas configuration: unknown
    pc file directory: /Users/reeschang/miniconda3/envs/ml4pxrds/lib/pkgconfig
    version: 3.9.0
  pybind11:
    detection method: pkgconfig
    include directory: /Users/reeschang/miniconda3/envs/ml4pxrds/include
    name: pybind11
    version: 2.12.0
Compilers:
  c:
    args: -march=core2, -mtune=haswell, -mssse3, -ftree-vectorize, -fPIC, -fstack-protector-strong,
      -O2, -pipe, -isystem, /Users/reeschang/miniconda3/envs/ml4pxrds/include, -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/scipy-split_1716470285601/work=/usr/local/src/conda/scipy-split-1.13.1,
      -fdebug-prefix-map=/Users/reeschang/miniconda3/envs/ml4pxrds=/usr/local/src/conda-prefix,
      -D_FORTIFY_SOURCE=2, -isystem, /Users/reeschang/miniconda3/envs/ml4pxrds/include,
      -mmacosx-version-min=10.13, -mmacosx-version-min=10.13
    commands: x86_64-apple-darwin13.4.0-clang
    linker: ld64
    linker args: -Wl,-headerpad_max_install_names, -Wl,-dead_strip_dylibs, -Wl,-rpath,/Users/reeschang/miniconda3/envs/ml4pxrds/lib,
      -L/Users/reeschang/miniconda3/envs/ml4pxrds/lib, -march=core2, -mtune=haswell,
      -mssse3, -ftree-vectorize, -fPIC, -fstack-protector-strong, -O2, -pipe, -isystem,
      /Users/reeschang/miniconda3/envs/ml4pxrds/include, -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/scipy-split_1716470285601/work=/usr/local/src/conda/scipy-split-1.13.1,
      -fdebug-prefix-map=/Users/reeschang/miniconda3/envs/ml4pxrds=/usr/local/src/conda-prefix,
      -D_FORTIFY_SOURCE=2, -isystem, /Users/reeschang/miniconda3/envs/ml4pxrds/include,
      -mmacosx-version-min=10.13, -mmacosx-version-min=10.13
    name: clang
    version: 16.0.6
  c++:
    args: -march=core2, -mtune=haswell, -mssse3, -ftree-vectorize, -fPIC, -fstack-protector-strong,
      -O2, -pipe, -stdlib=libc++, -fvisibility-inlines-hidden, -fmessage-length=0,
      -isystem, /Users/reeschang/miniconda3/envs/ml4pxrds/include, -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/scipy-split_1716470285601/work=/usr/local/src/conda/scipy-split-1.13.1,
      -fdebug-prefix-map=/Users/reeschang/miniconda3/envs/ml4pxrds=/usr/local/src/conda-prefix,
      -D_FORTIFY_SOURCE=2, -isystem, /Users/reeschang/miniconda3/envs/ml4pxrds/include,
      -mmacosx-version-min=10.13, -mmacosx-version-min=10.13
    commands: x86_64-apple-darwin13.4.0-clang++
    linker: ld64
    linker args: -Wl,-headerpad_max_install_names, -Wl,-dead_strip_dylibs, -Wl,-rpath,/Users/reeschang/miniconda3/envs/ml4pxrds/lib,
      -L/Users/reeschang/miniconda3/envs/ml4pxrds/lib, -march=core2, -mtune=haswell,
      -mssse3, -ftree-vectorize, -fPIC, -fstack-protector-strong, -O2, -pipe, -stdlib=libc++,
      -fvisibility-inlines-hidden, -fmessage-length=0, -isystem, /Users/reeschang/miniconda3/envs/ml4pxrds/include,
      -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/scipy-split_1716470285601/work=/usr/local/src/conda/scipy-split-1.13.1,
      -fdebug-prefix-map=/Users/reeschang/miniconda3/envs/ml4pxrds=/usr/local/src/conda-prefix,
      -D_FORTIFY_SOURCE=2, -isystem, /Users/reeschang/miniconda3/envs/ml4pxrds/include,
      -mmacosx-version-min=10.13, -mmacosx-version-min=10.13
    name: clang
    version: 16.0.6
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 3.0.10
  fortran:
    args: -march=core2, -mtune=haswell, -ftree-vectorize, -fPIC, -fstack-protector,
      -O2, -pipe, -isystem, /Users/reeschang/miniconda3/envs/ml4pxrds/include, -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/scipy-split_1716470285601/work=/usr/local/src/conda/scipy-split-1.13.1,
      -fdebug-prefix-map=/Users/reeschang/miniconda3/envs/ml4pxrds=/usr/local/src/conda-prefix
    commands: /Users/runner/miniforge3/conda-bld/scipy-split_1716470285601/_build_env/bin/x86_64-apple-darwin13.4.0-gfortran
    linker: ld64
    linker args: -Wl,-headerpad_max_install_names, -Wl,-dead_strip_dylibs, -Wl,-rpath,/Users/reeschang/miniconda3/envs/ml4pxrds/lib,
      -L/Users/reeschang/miniconda3/envs/ml4pxrds/lib, -march=core2, -mtune=haswell,
      -ftree-vectorize, -fPIC, -fstack-protector, -O2, -pipe, -isystem, /Users/reeschang/miniconda3/envs/ml4pxrds/include,
      -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/scipy-split_1716470285601/work=/usr/local/src/conda/scipy-split-1.13.1,
      -fdebug-prefix-map=/Users/reeschang/miniconda3/envs/ml4pxrds=/usr/local/src/conda-prefix
    name: gcc
    version: 12.3.0
  pythran:
    include directory: ../../_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.9/site-packages/pythran
    version: 0.16.0
Machine Information:
  build:
    cpu: x86_64
    endian: little
    family: x86_64
    system: darwin
  cross-compiled: false
  host:
    cpu: x86_64
    endian: little
    family: x86_64
    system: darwin
Python Information:
  path: /Users/reeschang/miniconda3/envs/ml4pxrds/bin/python
  version: '3.9'
```

**Comment by mdhaber** [comment]
Does it affect any other packages, e.g. `from scipy import interpolate`?

**Comment by rees-c** [comment]
Hey @mdhaber , I get the same error when I run `from scipy import interpolate` or `from scipy.interpolate import interp1d`

**Comment by mdhaber** [comment]
It sounds like an installation problem more than a bug in a particular sub-packages, but I've re-labled the issue issue because the `TypeError` manifests in `interpolate`.

**Comment by ev-br** [comment]
Looks like an installation-specific problem indeed. I guess we'd need to know 
- if the problem persists if you reinstall scipy into a fresh conda environment.
- how exactly you install scipy

**Comment by IcyFeather233** [comment]
I got same error when `import scipy.stats` in version 1.5.0

Install command: `pip install scikit-learn `

**Comment by lucascolley** [comment]
@IcyFeather233 could you clarify the SciPy version? 1.5.0 is *very* old at this point.

**Comment by IcyFeather233** [comment]
@lucascolley I cannot find a newer version
```
pip install scikit-learn==1.14.0
ERROR: Could not find a version that satisfies the requirement scikit-learn==1.14.0 (from versions: 0.9, 0.10, 0.11, 0.12, 0.12.1, 0.13, 0.13.1, 0.14, 0.14.1, 0.15.0, 0.15.1, 0.15.2, 0.16.0, 0.16.1, 0.17, 0.17.1, 0.18, 0.18.1, 0.18.2, 0.19.0, 0.19.1, 0.19.2, 0.20.0, 0.20.1, 0.20.2, 0.20.3, 0.20.4, 0.21.1, 0.21.2, 0.21.3, 0.22, 0.22.1, 0.22.2.post1, 0.23.0, 0.23.1, 0.23.2, 0.24.0, 0.24.1, 0.24.2, 1.0, 1.0.1, 1.0.2, 1.1.0, 1.1.1, 1.1.2, 1.1.3, 1.2.0rc1, 1.2.0, 1.2.1, 1.2.2, 1.3.0rc1, 1.3.0, 1.3.1, 1.3.2, 1.4.0rc1, 1.4.0, 1.4.1.post1, 1.4.2, 1.5.0rc1, 1.5.0)
ERROR: No matching distribution found for scikit-learn==1.14.0
```

**Comment by lucascolley** [comment]
Ah, that is `scikit-learn` 1.5.0, not SciPy 1.5.0.

After `pip install scikit-learn==1.5.0`, could you enter a Python interpreter and paste the result of `import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info); scipy.show_config()`?

**Comment by lucascolley** [comment]
as for the error message, `dfitpack` was touched for 1.14.0 by gh-20612, so a regression seems plausible. cc @rgommers @thalassemia

**Comment by IcyFeather233** [comment]
> Ah, that is `scikit-learn` 1.5.0, not SciPy 1.5.0.
> 
> After `pip install scikit-learn==1.5.0`, could you enter a Python interpreter and paste the result of `import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info); scipy.show_config()`? Or does the error arise immediately from that?

```
1.13.1 1.26.4 sys.version_info(major=3, minor=9, micro=19, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: pkgconfig
    found: true
    include directory: /opt/arm64-builds/include
    lib directory: /opt/arm64-builds/lib
    name: openblas
    openblas configuration: USE_64BITINT=0 DYNAMIC_ARCH=1 DYNAMIC_OLDER= NO_CBLAS=
      NO_LAPACK= NO_LAPACKE= NO_AFFINITY=1 USE_OPENMP= SANDYBRIDGE MAX_THREADS=64
    pc file directory: /opt/arm64-builds/lib/pkgconfig
    version: 0.3.27
  lapack:
    detection method: pkgconfig
    found: true
    include directory: /opt/arm64-builds/include
    lib directory: /opt/arm64-builds/lib
    name: openblas
    openblas configuration: USE_64BITINT=0 DYNAMIC_ARCH=1 DYNAMIC_OLDER= NO_CBLAS=
      NO_LAPACK= NO_LAPACKE= NO_AFFINITY=1 USE_OPENMP= SANDYBRIDGE MAX_THREADS=64
    pc file directory: /opt/arm64-builds/lib/pkgconfig
    version: 0.3.27
  pybind11:
    detection method: config-tool
    include directory: unknown
    name: pybind11
    version: 2.12.0
Compilers:
  c:
    commands: cc
    linker: ld64
    name: clang
    version: 15.0.0
  c++:
    commands: c++
    linker: ld64
    name: clang
    version: 15.0.0
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 3.0.10
  fortran:
    commands: gfortran
    linker: ld64
    name: gcc
    version: 12.1.0
  pythran:
    include directory: ../../../../../../private/var/folders/2f/8t5k6yr535sdw0s4glnpxrzm0000gn/T/pip-build-env-ddcl7dz8/overlay/lib/python3.9/site-packages/pythran
    version: 0.15.0
Machine Information:
  build:
    cpu: aarch64
    endian: little
    family: aarch64
    system: darwin
  cross-compiled: false
  host:
    cpu: aarch64
    endian: little
    family: aarch64
    system: darwin
Python Information:
  path: /private/var/folders/2f/8t5k6yr535sdw0s4glnpxrzm0000gn/T/cibw-run-_36keccq/cp39-macosx_arm64/build/venv/bin/python
  version: '3.9'
```

**Comment by lucascolley** [comment]
Are you able to `pip install scipy==1.14.0` now? If so, does that make the error go away, or is it still there?

**Comment by IcyFeather233** [comment]
```
 pip install scipy==1.14.0       
ERROR: Ignored the following yanked versions: 1.11.0
ERROR: Ignored the following versions that require a different python version: 1.14.0 Requires-Python >=3.10; 1.14.0rc1 Requires-Python >=3.10; 1.14.0rc2 Requires-Python >=3.10
ERROR: Could not find a version that satisfies the requirement scipy==1.14.0 (from versions: 0.8.0, 0.9.0, 0.10.0, 0.10.1, 0.11.0, 0.12.0, 0.12.1, 0.13.0, 0.13.1, 0.13.2, 0.13.3, 0.14.0, 0.14.1, 0.15.0, 0.15.1, 0.16.0, 0.16.1, 0.17.0, 0.17.1, 0.18.0, 0.18.1, 0.19.0, 0.19.1, 1.0.0, 1.0.1, 1.1.0, 1.2.0, 1.2.1, 1.2.2, 1.2.3, 1.3.0, 1.3.1, 1.3.2, 1.3.3, 1.4.0, 1.4.1, 1.5.0, 1.5.1, 1.5.2, 1.5.3, 1.5.4, 1.6.0, 1.6.1, 1.6.2, 1.6.3, 1.7.0, 1.7.1, 1.7.2, 1.7.3, 1.8.0rc1, 1.8.0rc2, 1.8.0rc3, 1.8.0rc4, 1.8.0, 1.8.1, 1.9.0rc1, 1.9.0rc2, 1.9.0rc3, 1.9.0, 1.9.1, 1.9.2, 1.9.3, 1.10.0rc1, 1.10.0rc2, 1.10.0, 1.10.1, 1.11.0rc1, 1.11.0rc2, 1.11.1, 1.11.2, 1.11.3, 1.11.4, 1.12.0rc1, 1.12.0rc2, 1.12.0, 1.13.0rc1, 1.13.0, 1.13.1)
ERROR: No matching distribution found for scipy==1.14.0
```

**Comment by lucascolley** [comment]
Ah you're still on Python 3.9. Any reason you can't upgrade to >=3.10?

**Comment by IcyFeather233** [comment]
update python to 3.12 and it works without error, thanks!

**Comment by lucascolley** [comment]
@rees-c please upgrade Python to >=3.10 and SciPy to 1.14.0, and please reopen this issue if the error persists 👍

**Comment by ledinhtri97** [comment]
I had same error. from python3.10

```
>>> import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info); scipy.show_config()
1.14.0 1.26.4 sys.version_info(major=3, minor=10, micro=12, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: pkgconfig
    found: true
    include directory: /opt/_internal/cpython-3.10.13/lib/python3.10/site-packages/scipy_openblas32/include
    lib directory: /opt/_internal/cpython-3.10.13/lib/python3.10/site-packages/scipy_openblas32/lib
    name: scipy-openblas
    openblas configuration: OpenBLAS 0.3.27.dev DYNAMIC_ARCH NO_AFFINITY Zen MAX_THREADS=64
    pc file directory: /project
    version: 0.3.27.dev
  lapack:
    detection method: pkgconfig
    found: true
    include directory: /opt/_internal/cpython-3.10.13/lib/python3.10/site-packages/scipy_openblas32/include
    lib directory: /opt/_internal/cpython-3.10.13/lib/python3.10/site-packages/scipy_openblas32/lib
    name: scipy-openblas
    openblas configuration: OpenBLAS 0.3.27.dev DYNAMIC_ARCH NO_AFFINITY Zen MAX_THREADS=64
    pc file directory: /project
    version: 0.3.27.dev
  pybind11:
    detection method: config-tool
    include directory: unknown
    name: pybind11
    version: 2.12.0
Compilers:
  c:
    commands: cc
    linker: ld.bfd
    name: gcc
    version: 10.2.1
  c++:
    commands: c++
    linker: ld.bfd
    name: gcc
    version: 10.2.1
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 3.0.10
  fortran:
    commands: gfortran
    linker: ld.bfd
    name: gcc
    version: 10.2.1
  pythran:
    include directory: ../../tmp/pip-build-env-vlr41q3a/overlay/lib/python3.10/site-packages/pythran
    version: 0.16.1
Machine Information:
  build:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
  cross-compiled: false
  host:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
Python Information:
  path: /opt/python/cp310-cp310/bin/python
  version: '3.10'
 ```

**Comment by lucascolley** [comment]
Thanks @ledinhtri97 , could you post any details you have in gh-21546?

**Comment by ledinhtri97** [comment]
@lucascolley I just solved. it worked on docker 26.1.0. and have issue on latest docker version.
I try to downgrade docker image then everything fine.
thank you

**Comment by ay27** [comment]
I also encountered this error with Python 3.10 and SciPy 1.14.0. I solved it by downgrading SciPy to version 1.12.x:

```
pip install 'scipy<1.13.0'
```

**Comment by sukui-genomics-cn** [comment]
> @rees-c please upgrade Python to >=3.10 and SciPy to 1.14.0, and please reopen this issue if the error persists 👍

I get same error with python=3.10 and SciPy=1.14.0

```markdown
from scipy.stats import pearsonr, spearmanr
```

This is my error. can you give me some advice.

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/share/huadjyin/home/s_sukui/.conda/envs/jamba_py10/lib/python3.10/site-packages/scipy/stats/__init__.py", line 610, in <module>
    from ._stats_py import *
  File "/home/share/huadjyin/home/s_sukui/.conda/envs/jamba_py10/lib/python3.10/site-packages/scipy/stats/_stats_py.py", line 49, in <module>
    from . import distributions
  File "/home/share/huadjyin/home/s_sukui/.conda/envs/jamba_py10/lib/python3.10/site-packages/scipy/stats/distributions.py", line 10, in <module>
    from . import _continuous_distns
  File "/home/share/huadjyin/home/s_sukui/.conda/envs/jamba_py10/lib/python3.10/site-packages/scipy/stats/_continuous_distns.py", line 12, in <module>
    from scipy.interpolate import BSpline
  File "/home/share/huadjyin/home/s_sukui/.conda/envs/jamba_py10/lib/python3.10/site-packages/scipy/interpolate/__init__.py", line 167, in <module>
    from ._interpolate import *
  File "/home/share/huadjyin/home/s_sukui/.conda/envs/jamba_py10/lib/python3.10/site-packages/scipy/interpolate/_interpolate.py", line 12, in <module>
    from . import _fitpack_py
  File "/home/share/huadjyin/home/s_sukui/.conda/envs/jamba_py10/lib/python3.10/site-packages/scipy/interpolate/_fitpack_py.py", line 8, in <module>
    from ._fitpack_impl import bisplrep, bisplev, dblint  # noqa: F401
  File "/home/share/huadjyin/home/s_sukui/.conda/envs/jamba_py10/lib/python3.10/site-packages/scipy/interpolate/_fitpack_impl.py", line 103, in <module>
    'iwrk': array([], dfitpack_int), 'u': array([], float),
TypeError

**Comment by LiWeitu** [comment]
@sukui-genomics-cn 
I encountered the same Error with python=3.10, SciPy=1.14.1 and numpy=1.26.4 on Ubuntu 18.04.6, I finally solved it by downgrading SciPy to version 1.13.0. Maybe you can try this.

But the same code and the same environment(python=3.10, SciPy=1.14.1 and numpy=1.26.4) work normally on Windows.
